### PR TITLE
DNS name error message improvement

### PR DESCRIPTION
**Pull request by ryanmcnamara**
**What this PR does / why we need it**:
Small error message fix. The error message misled me slightly / would have saved time if correct

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
n/a
**Special notes for your reviewer**:
n/a
**Release note**:

n/a

### CODE REVIEW

**Comment by sttts** [comment]
Some native speaker should review this.

File: staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go [hunk 1]
```diff
@@ -126,7 +126,7 @@ func IsDNS1123Subdomain(value string) []string {
 }
 
 const dns1035LabelFmt string = "[a-z]([-a-z0-9]*[a-z0-9])?"
-const dns1035LabelErrMsg string = "a DNS-1035 label must consist of lower case alphanumeric characters or '-', and must start and end with an alphanumeric character"
+const dns1035LabelErrMsg string = "a DNS-1035 label must consist of lower case alphanumeric characters or '-', start with an alphabetic character, and end with an alphanumeric character"
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go]: This reads slightly differently.  I don't really care one way or the other.

@adellape recommendation on phrasing for a terse-ish api level error message?

**Comment by deads2k** [comment]
ah, I see

/lgtm
/approve no-issue
/release-note-none

**Comment by deads2k** [comment]
@ryanmcnamara Can you sign the cla.

File: staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go (hunk 1 above)
Comment (adellape) [inline staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go]: If the distinction is that it cannot start with a numeric character, I don't have a better suggestion.

**Comment by ryanmcnamara** [comment]
@adellape yup that's the distinction
@deads2k working on it, my organization needs to sign first, which should be in the works. Sorry for the delay

**Comment by ryanmcnamara** [comment]
I've now signed the cla as an individual. Sorry for the delay

**Pull request merged**